File: frontend/src/types/session.ts
export interface Session {
  id: string;
  organizationId: string;
  userId: string;
  title: string | null;
  hostIds: string[];
  createdAt: string;
  lastMessageAt: string | null;
  isJobRunSession: boolean;
  totalTokens: number;
  totalCost: number;
}

export interface Message {
  id: string;
  sessionId: string;
  role: "user" | "assistant";
  content: string;
  toolCallsJson: string | null;
  tokensUsed: number | null;
  createdAt: string;
}

export interface SendMessageRequest {
  content: string;
  hostIds?: string[];
  model?: string;
}

export interface CreateSessionRequest {
  title?: string;
  hostIds?: string[];
}

export interface ToolCall {
  id: string;
  name: string;
  arguments: Record<string, unknown>;
  result?: string;
}

File: src/InfraLLM.Core/Models/Session.cs
namespace InfraLLM.Core.Models;

public class Session
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string? Title { get; set; }
    public List<Guid> HostIds { get; set; } = [];
    public DateTime CreatedAt { get; set; }
    public DateTime? LastMessageAt { get; set; }
    public bool IsJobRunSession { get; set; }
    public int TotalTokens { get; set; }
    public decimal TotalCost { get; set; }

    public Organization Organization { get; set; } = null!;
    public List<Message> Messages { get; set; } = [];
}
